vividus-plugin-azure-resource-manager/src/main/java/org/vividus/azure/configuration/AzureKeyVaultPropertiesProcessor.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.azure.configuration;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+import com.azure.core.http.HttpMethod;
+import com.azure.core.management.AzureEnvironment;
+import com.azure.core.management.profile.AzureProfile;
+
+import org.apache.commons.lang3.Validate;
+import org.vividus.azure.client.AzureResourceManagerClient;
+import org.vividus.azure.identity.CredentialFactory;
+import org.vividus.configuration.AbstractPropertiesProcessor;
+import org.vividus.util.json.JsonPathUtils;
+
+public class AzureKeyVaultPropertiesProcessor extends AbstractPropertiesProcessor
+{
+    private static final String KEY_VAULT_PROCESSOR_ENABLED_PROPERTY = "azure.key-vault.processor.enabled";
```

Comment:
why is it needed?

Reply:
**Precondition:** we have access to particular KeyVault instance with particular VPN. 
**Use case:** issue with VPN -> will block execution tests at all. With this property we will have ability to override only needed properties for execution without calls to AKV.

Reply:
if properties are overridden, then there should be no properties with AKV placeholder, and no AKV invocation are expected, so the problem statement is not clear

Reply:
discussed and updated